runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -696,7 +696,17 @@ public synchronized void invalidateLayout() {
             // Don't create a new request for a layout if there is one pending.
             return;
         }
-        layout = fetchLayout(CFUtils.getUninterruptibly(layout).getLayoutServers());
+        List<String> lastKnownLayoutServers = layoutServers;
+        try {
+            if (layout != null) {
+                // Layout fetch can result in an exception if the systemDownHandler was invoked and
+                // the completable future was completed exceptionally.
+                lastKnownLayoutServers = CFUtils.getUninterruptibly(layout).getLayoutServers();
+            }
+        } catch (Throwable th) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)